src/envoy/utils/filter_state_utils.cc

```diff
@@ -44,6 +54,25 @@ absl::string_view getStringFilterState(
   return filter_state.getDataReadOnly<StringAccessor>(data_name).asString();
 }
 
+void setErrorFilterState(Envoy::StreamInfo::FilterState& filter_state,
+                         const google::rpc::Status& status) {
```

Comment:
maybe pass in rpc::Status&& so it can be either swap-ed or moved?  try to see if we can avoid copy

Reply:
Done, but I had to duplicate code. What do you think?